src/jemalloc.c

```diff
@@ -2631,7 +2631,11 @@ isfree(tsd_t *tsd, void *ptr, size_t usize, tcache_t *tcache, bool slow_path) {
 			rtree_szind_slab_read(tsd_tsdn(tsd), &extents_rtree,
 			    rtree_ctx, (uintptr_t)ptr, true, &ctx->szind,
 			    &ctx->slab);
-			assert(ctx->szind == sz_size2index(usize));
+			bool sz_correct = (ctx->szind == sz_size2index(usize));
+			if (config_opt_safety_checks && !sz_correct) {
+				safety_check_fail_sized_dealloc();
+			}
+			assert(sz_correct);
```

Comment:
config_opt_safety_checks is true whenever asserts are enabled; this can't ever fire.

Reply:
It looks to be controlled separately, through `--enable-opt-safety-checks` (which is default off upstream https://github.com/jemalloc/jemalloc/blob/dev/configure.ac#L1475)?

Reply:
It's set with the bool rather than the macro -- https://github.com/jemalloc/jemalloc/blob/21cfe59ff7b10a61dabe26cd3dbfb7a255e1f5e8/include/jemalloc/internal/jemalloc_preamble.h.in#L171

Reply:
Oh I see what you mean. Will update.